### PR TITLE
Update doc and add check for unbounded action space

### LINKED ISSUE #897: [Bug] Unbounded action spaces
### Bug, feature, reasonable design choice - underline your opinion

The RL algorithms implemented in `stable-baselines3` do not support unbounded action spaces, i.e. 

```
gym.spaces.Box(shape=some_shape, low=-np.inf, high=np.inf)
```

### 🐛 Definitely a Bug

When someone attempts using stable-baselines on an environment with an unbounded action space, the failure happens silently. There is no error of any kind as the "learning" goes on for however many timesteps you have specified, the only problem is that all tensors in the resulting neural networks become `float('nan')` tensors. I also have not found it mentioned explicitly anywhere in the documentation, so the user has no way to learn that they are using an unsupported environment until many steps into the training and even then it takes some debugging to figure out what was the reason.

### To Reproduce

```python
import numpy as np
from stable_baselines3 import TD3
from stable_baselines3.common.monitor import Monitor
import gym

class MirrorEnv(gym.Env):
  def __init__(self):
        self.observation_space = gym.spaces.Box(shape=some_shape, 
                                                low=-np.inf, high=np.inf, 
                                                dtype=np.float32)
        self.action_space = gym.spaces.Box(shape=some_shape, 
                                               low=-np.inf, high=np.inf,
                                               dtype=np.float32)

  def reset(self):
     return self.observation_space.sample()

  def step(self, a):
     obs = a
     r = np.mean(a)
     return obs, r, False, None

env = MirrorEnv()
env = Monitor(env)

model = TD3('MlpPolicy', env, verbose=1)
model.learn(total_timesteps=25000)
```

```
---------------------------------
| rollout/           |          |
|    ep_len_mean     | 1e+03    |
|    ep_rew_mean     | nan      |
| time/              |          |
|    episodes        | 4        |
|    fps             | 0        |
|    time_elapsed    | 4411     |
|    total_timesteps | 4000     |
| train/             |          |
|    actor_loss      | nan      |
|    critic_loss     | nan      |
|    learning_rate   | 0.001    |
|    n_updates       | 3000     |
---------------------------------
```

### Expected behavior

Either: 

```
---------------------------------
| rollout/           |          |
|    ep_len_mean     | 1e+03    |
|    ep_rew_mean     | 42       |
| time/              |          |
|    episodes        | 4        |
|    fps             | 0        |
|    time_elapsed    | 4411     |
|    total_timesteps | 4000     |
| train/             |          |
|    actor_loss      | 42       |
|    critic_loss     | 42       |
|    learning_rate   | 0.001    |
|    n_updates       | 3000     |
---------------------------------
```

(42 is used to represent 'some number' without loss of generality)

Or:

```
ValueError("Unsupported action space: make sure your Box space has a finite lower and upper bound")
```

### System Info

```
OS: Linux-4.15.0-176-generic-x86_64-with-glibc2.27 #185-Ubuntu SMP Tue Mar 29 17:40:04 UTC 2022
Python: 3.9.12
Stable-Baselines3: 1.5.0
PyTorch: 1.11.0+cu102
GPU Enabled: True
Numpy: 1.22.3
Gym: 0.21.0
```

### Additional context

This hints, of course, at a much bigger issue in the field of Reinforcement Learning that there is implicit knowledge like "unbounded action spaces are technically allowed by gym, but they aren't a thing in modern RL methods" that exists in everyone's heads, but not in the documentation of our tools.

### Checklist

- [x] I have checked that there is no similar [issue](https://github.com/DLR-RM/stable-baselines3/issues) in the repo (**required**)
- [x] I have read the [documentation](https://stable-baselines3.readthedocs.io/en/master/) (**required**)
- [x] I have provided a minimal working example to reproduce the bug (**required**)

### PR DESCRIPTION
**Pull request by araffin**
## Motivation and Context

- [x] I have raised an issue to propose this change ([required](https://github.com/DLR-RM/stable-baselines3/blob/master/CONTRIBUTING.md) for new features and bug fixes)

closes #897 

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [x] Documentation (update in the documentation)

## Checklist:

- [x] I've read the [CONTRIBUTION](https://github.com/DLR-RM/stable-baselines3/blob/master/CONTRIBUTING.md) guide (**required**)
- [x] I have updated the changelog accordingly (**required**).
- [x] My change requires a change to the documentation.
- [x] I have updated the tests accordingly (*required for a bug fix or a new feature*).
- [x] I have updated the documentation accordingly.
- [x] I have reformatted the code using `make format` (**required**)
- [x] I have checked the codestyle using `make check-codestyle` and `make lint` (**required**)
- [x] I have ensured `make pytest` and `make type` both pass. (**required**)
- [x] I have checked that the documentation builds using `make doc` (**required**)

Note: You can run most of the checks using `make commit-checks`.

Note: we are using a maximum length of 127 characters per line

### CODE REVIEW

File: stable_baselines3/common/base_class.py [hunk 1]
```diff
@@ -185,6 +185,11 @@ def __init__(
             if self.use_sde and not isinstance(self.action_space, gym.spaces.Box):
                 raise ValueError("generalized State-Dependent Exploration (gSDE) can only be used with continuous actions.")
 
+            if isinstance(self.action_space, gym.spaces.Box):
+                assert np.all(
+                    np.isfinite(np.array([self.action_space.low, self.action_space.high]))
+                ), "Continuous action space must have a finite lower and upper bound"
```
Comment (Miffyli) [inline stable_baselines3/common/base_class.py:191]: Maybe a pointer to the doc about this (here? https://stable-baselines3.readthedocs.io/en/master/guide/rl_tips.html#tips-and-tricks-when-creating-a-custom-environment ). It is not exact answer but healthy tips how env should be made :).

Other than that LGTM!

Comment (araffin) [reply stable_baselines3/common/base_class.py:191]: hmm, I think the error message is clear enough of what is wrong, and the env checker should do the rest ;)
